packages/split-view/src/SplitView.ts

```diff
@@ -0,0 +1,416 @@
+        ];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public vertical = false;
+
+    @property({ type: Boolean, reflect: true })
+    public resizable = false;
+
+    @property({ type: Boolean, reflect: true })
+    public collapsible = false;
+
+    /** The minimum size of the primary pane */
+    @property({ type: Number, attribute: 'primary-min' })
+    public primaryMin = 304;
+
+    /** The maximum size of the primary pane */
+    @property({ type: Number, attribute: 'primary-max' })
+    public primaryMax = Infinity;
+
+    /** The size of the primary pane */
+    @property({ type: Number, attribute: 'primary-size' })
+    public primarySize?: number;
+
+    /** The default size of the primary pane */
+    @property({ type: Number, attribute: 'primary-default' })
+    public primaryDefault = 304;
+
+    /** The minimum size of the secondary pane */
+    @property({ type: Number, attribute: 'secondary-min' })
+    public secondaryMin = 304;
+
+    /** The maximum size of the secondary pane */
+    @property({ type: Number, attribute: 'secondary-max' })
+    public secondaryMax = Infinity;
+
+    @property({ type: Number, attribute: false })
+    public dividerPosition = this.primaryDefault;
+
+    @property({ type: Number, attribute: false })
+    public minPos = 0;
+
+    @property({ type: Number, attribute: false })
+    public maxPos = Infinity;
+
+    @property({ type: Boolean, reflect: true, attribute: 'is-collapsed-start' })
+    public isCollapsedStart = this.dividerPosition === 0;
+
+    @property({ type: Boolean, reflect: true, attribute: 'is-collapsed-end' })
+    public isCollapsedEnd = false;
+
+    @property({ type: Boolean, reflect: true })
+    public dragging = false;
+
+    @property({ type: Boolean, reflect: true })
+    public hovered = false;
+
+    @property()
+    public label?: string;
+
+    private offset = 0;
+    private size = 0;
+    private isOver = false;
+    private lastPosition = 0;
+
+    public constructor() {
+        super();
+        this.onPointerDragged = this.onPointerDragged.bind(this);
+        this.onPointerUp = this.onPointerUp.bind(this);
+        this.resize = this.resize.bind(this);
+        this.updatePosition = this.updatePosition.bind(this);
+        this.getSplitterSize = this.getSplitterSize.bind(this);
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        window.addEventListener('resize', this.resize);
+    }
+
+    public disconnectedCallback(): void {
+        window.removeEventListener('resize', this.resize);
+        super.disconnectedCallback();
+    }
+
+    protected render(): TemplateResult {
+        const dimension = this.vertical
+            ? ORIENTATIONS.vertical
+            : ORIENTATIONS.horizontal;
+
+        return html`
+            <div
+                class="pane"
+                style=${`${dimension}: ${this.dividerPosition}px`}
+            >
+                <slot name="primary"></slot>
```

Comment:
I can be convinced that this is the right thing, but I'm not excited with how it complicates the external API here by adding two `slot`s for a user to think about. I've just pushed a branch with the approach I discussed last week `westbrook/splitview`. Thoughts?

Reply:
I used your slot css solution now, works perfect :-)